client/credentials_test.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"testing"
+	"strconv"	
+)
+
+func TestCredentials(t *testing.T){
+
+	cred := &Credentials{
+		ClientID          : "tester"
+		AccessToken       : "no-secret"
+	}
+
+	var credentialsTests = []struct {
+		meth string
+		url  string
+		host string
+		port int
+		hdr  string
+		now  int64
+		perr error
+		verr error
+		key  string
+		hash func() hash.Hash
+		rply bool
```

Comment:
I would try to avoid names like this.. Just use  `method`, `reply`, `header`, they aren't that long, I understand how `verr` makes sense..